ui/test/nuxeo-tooltip.test.js

```diff
@@ -0,0 +1,48 @@
+/**
+@license
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import { html, fixture, flush, isElementVisible, timePasses } from '@nuxeo/testing-helpers';
+import '../widgets/nuxeo-tooltip.js';
+
+suite('nuxeo-tooltip', async () => {
+  test('Should not show a paper-tooltip if it has hidden atrribute set to true', async () => {
+    const tooltip = await fixture(
+      html`
+        <nuxeo-tooltip hidden="true">Hello</nuxeo-tooltip>
+      `,
+    );
+    await flush();
+    tooltip.show();
+    await timePasses(500);
+
+    const paperTooltip = document.querySelector('paper-tooltip');
+    expect(isElementVisible(paperTooltip)).to.be.false;
+  });
+
+  test('Should show a paper-tooltip if no hidden attribute is set', async () => {
+    const tooltip = await fixture(
+      html`
+        <nuxeo-tooltip>Hello</nuxeo-tooltip>
+      `,
+    );
+    await flush();
+    tooltip.show();
+    await timePasses(500);
```

Comment:
It seems that our `nuxeo-tooltip` has an animation delay defaulted to 500ms, but I wouldn't trust on that for the test it self. Also, waiting X amount of time may turn the test flaky (it would depend on environment that is running the tests).

Since we know that we are adding a DOM node to `document.body` I would suggest relying on `waitForChildListMutation`.

Reply:
the problem is that after the node is added, what is taking time is making its class hidden get removed. I am trying waitChanged on the paper-element but having issues making it work.